src/utils/builder.ts

```diff
@@ -0,0 +1,297 @@
+}
+
+export function addGuidanceStateCql(pathway: Pathway, key: string, cql: string): void {
+  pathway.states[key].cql = cql;
+}
+
+// TODO: possibly add more action methods
+export function addAction(
+  pathway: Pathway,
+  key: string,
+  type: string,
+  description: string,
+  resource: MedicationRequest | ServiceRequest
+): string {
+  const id = shortid.generate();
+  const action = {
+    id: id,
+    type: type,
+    description: description,
+    resource: resource
+  };
+  pathway.states[key].action.push(action);
+  return id;
+}
+
+/*
+Update Element Functions
+*/
+export function updatePathwayName(pathway: Pathway, name: string): void {
+  pathway.name = name;
+}
+
+export function updatePathwayDescription(pathway: Pathway, description: string): void {
+  pathway.description = description;
+}
+
+export function updateTransition(
+  pathway: Pathway,
+  startStateKey: string,
+  endStateKey: string,
+  transitionId: string
+): void {
+  const transition: Transition = pathway.states[startStateKey].transitions.find(
+    (transition: Transition) => transition.id === transitionId
+  );
+  transition.transition = endStateKey;
+}
+
+export function updateTransitionConditionDescription(
+  pathway: Pathway,
+  startNodeKey: string,
+  transitionId: string,
+  description: string
+): void {
+  const foundTransition: Transition | null = pathway.states[startNodeKey].transitions.find(
+    (transition: Transition) => transition.id === transitionId
+  );
+
+  if (foundTransition?.condition) foundTransition.condition.description = description;
+}
+
+export function updateTransitionConditionCql(
+  pathway: Pathway,
+  startNodeKey: string,
+  transitionId: string,
+  cql: string
+): void {
+  const foundTransition: Transition | null = pathway.states[startNodeKey].transitions.find(
+    (transition: Transition) => transition.id === transitionId
+  );
+
+  if (foundTransition?.condition) foundTransition.condition.cql = cql;
+}
+
+export function updateActionType(
+  pathway: Pathway,
+  stateKey: string,
+  actionId: string,
+  type: string
+): void {
+  if (pathway.states[stateKey].action) {
+    const action = pathway.states[stateKey].action.find((action: Action) => action.id === actionId);
+    action.type = type;
+  }
+}
+
+export function updateActionDescription(
+  pathway: Pathway,
+  stateKey: string,
+  actionId: string,
+  description: string
+): void {
+  if (pathway.states[stateKey].action) {
+    const action = pathway.states[stateKey].action.find((action: Action) => action.id === actionId);
+    action.description = description;
+  }
+}
+
+export function updateActionResource(
+  pathway: Pathway,
+  stateKey: string,
+  actionId: string,
+  resource: MedicationRequest | ServiceRequest
+): void {
+  if (pathway.states[stateKey].action) {
+    const action = pathway.states[stateKey].action.find((action: Action) => action.id === actionId);
+    action.resource = resource;
+  }
+}
+
```

Comment:
We need a way to switch a branch node to an action node and an action node to a branch node.

Reply:
Added